src/utils/async.ts

```diff
@@ -1,4 +1,78 @@
 import { reportError } from './errors';
+import { SpawnOptions } from 'child_process';
+import { logger } from '../logger';
+import { isDryRun } from './helpers';
+import { SpawnProcessOptions, spawnProcess } from './system';
+
+/**
+ * Delay between retries of running commands, in seconds.
+ */
+const RETRY_DELAY_SECS = 3;
+
+/**
+ * Exponential backoff applied to the retry delay.
+ */
+const RETRY_EXP_FACTOR = 2;
+
+/**
+ * Maximum number of retries including the initial one when a command fails.
+ */
+const MAX_RETRIES = 5;
+
+/**
+ * Retrying options for retrySpawnProcess()
+ */
+export interface RetryOptions {
+  maxRetries?: number;
+  retryDelay?: number;
+  retryExpFactor?: number;
+}
+
+/**
+ * Retries executing the command, delaying ongoing calls for some time.
+ *
+ * After every call to the function, a delay in seconds is present before the
+ * next call, and this is incremented exponentially by the factor (see
+ * retry options).
+ *
+ * @param command the command to run.
+ * @param args optional arguments to pass to the command.
+ * @param retryOptions optional options to configure retry delays and retries.
+ * @param spawnOptions optional options to pass to child_process.spawn
+ * @param spawnProcessOptions optional options to pass to spawnProcess()
+ * @returns a promise that resolves when command exits successfully.
+ * @async
+ */
+export async function retrySpawnProcess(
+  command: string,
+  args: string[] = [],
+  retryOptions: RetryOptions = {},
+  spawnOptions: SpawnOptions = {},
+  spawnProcessOptions: SpawnProcessOptions = {}
+): Promise<Buffer | undefined> {
+  const argsString = args.map(arg => `"${arg}"`).join(' ');
+
+  if (isDryRun() && !spawnProcessOptions.enableInDryRunMode) {
+    logger.info('[dry-run] Not spawning process:', `${command} ${argsString}`);
+    return undefined;
+  }
+
+  const maxRetries = retryOptions.maxRetries || MAX_RETRIES;
```

Comment:
This would use `MAX_RETRIES` when one sets `retryOptions.maxRetries` to `0`.
```suggestion
  const maxRetries = retryOptions.maxRetries ?? MAX_RETRIES;
```